### DISCUSSION TITLE
Are the tests inside the test/orm directory meant to be run only with the SQLite dialect?

### DISCUSSION THREAD

**Discussion by ybjeon01** [sqlalchemy/sqlalchemy#11935] [Usage Questions] [answered]
Hello, I am currently developing a new dialect and, in addition to running the tests in the testing/suite, I’ve also attempted to run the tests in test/orm. However, I noticed that many of these tests are failing. To troubleshoot, I ran the test/orm tests using SQLAlchemy’s built-in Oracle dialect in my local environment. While only three tests failed with in-memory SQLite, hundreds of tests failed with Oracle. This makes me think I may have misunderstood something or executed the tests incorrectly. Could you help me understand what might have gone wrong? Or are the tests outside of test/dialects specifically designed to run with the SQLite dialect?

Thank you

The following is a Python script I created to make debugging easier in my IDE. While debugging, I’ve been studying SQLAlchemy. I haven't modified any files like setup.cfg or requirements.py.

```Python3
import pytest
import os

def main():

    # result = pytest.main(["--db", "docker_oracle", "test/orm"])
    result = pytest.main(["--db", "default", "-n", "4", "test/orm"])

    if result == 0:
        print("All tests passed.")
    else:
        print(f"Tests failed with code: {result}")

if __name__ == "__main__":
    main()
```

Oracle image: gvenzl/oracle-free:23-slim
Python version: 3.12.3
Pip list:
```
Package                Version  Editable project location
---------------------- -------- ------------------------------------------------------------
asttokens              2.4.1
cffi                   1.17.1
cryptography           43.0.1
cx_Oracle              8.3.0
decorator              5.1.1
execnet                2.1.1
executing              2.1.0
greenlet               3.1.1
iniconfig              2.0.0
ipython                8.27.0
jedi                   0.19.1
matplotlib-inline      0.1.7
mysql-connector-python 9.0.0
oracledb               2.4.1
packaging              24.1
parso                  0.8.4
pexpect                4.9.0
pip                    24.2
pluggy                 1.5.0
prompt_toolkit         3.0.48
ptyprocess             0.7.0
pure_eval              0.2.3
pycparser              2.22
Pygments               2.18.0
PyMySQL                1.1.1
pyodbc                 5.1.0
pytest                 8.3.3
pytest-xdist           3.6.1
six                    1.16.0
SQLAlchemy             2.0.35
stack-data             0.6.3
traitlets              5.14.3
typing_extensions      4.12.2
wcwidth                0.2.13

```

**Comment by zzzeek** [comment, marked as answer]
the test/orm tests are meant to be run with SQLAlchemy's own internal dialects in particular ways - if you are writing a third party dialect, the test suite that you want to target is in https://github.com/sqlalchemy/sqlalchemy/tree/main/lib/sqlalchemy/testing/suite .  There's a specific documented way to do this which you should review in https://github.com/sqlalchemy/sqlalchemy/blob/main/README.dialects.rst .  For a real world example of how this looks you can see the sqlalchemy-access dialect at https://github.com/gordthompson/sqlalchemy-access .

so you dont want to target anything in test/ basically.

**Reply by ybjeon01** [reply]
Thank you!!

**Reply by ybjeon01** [reply]
@zzzeek,  When you mention 'internal dialects,' do you mean the dialects that are provided by SQLAlchemy by default? If you don't mind, could you point me to the files where I can see how it operates in the particular ways you're referring to? I'm really interested in SQLAlchemy's code and would like to study its internal workings further.

**Reply by gordthompson** [reply]
Look in the folder `lib/sqlalchemy/dialects`

**Reply by ybjeon01** [reply]
Yes, I found `lib/sqlalchemy/dialects` and read some code in oracle dialect, but I'm not sure which internal dialects are used for the tests in the `/test` folder, excluding the `test/dialect` folder. From what I understand, the tests in the `/test` folder, apart from the `test/dialect` folder, are executed using only the default dialect specified in `setup.cfg`, which is in-memory SQLite. Am I understanding correctly?

**Reply by CaselIT** [reply]
No. You can look at the pipeline on jenkins, for example https://jenkins.sqlalchemy.org/job/sqlalchemy_21_gerrit/766/

Most dialect are run with the backend-only flag

**Reply by ybjeon01** [reply]
Thank you!
